src/evict/evict_page.c

```diff
@@ -453,6 +455,16 @@ __evict_child_check(WT_SESSION_IMPL *session, WT_REF *parent)
 			if (__wt_page_del_active(session, child, true))
 				return (EBUSY);
 			break;
+		case WT_REF_LOOKASIDE:
+			/*
+			 * Check if the lookaside history is obsolete, which
+			 * will flip the state to WT_REF_DISK.
+			 */
+			__wt_las_page_obsolete_check(session, child);
+			if (child->state != WT_REF_DISK &&
+			    child->state != WT_REF_DELETED)
+				return (EBUSY);
+			break;
```

Comment:
`will flip the state to WT_REF_DISK.` -> `will flip the state to WT_REF_DISK or WT_REF_DELETED.`

Reply:
In the case of the page reverting to `WT_REF_DELETED`, isn't there some chance that calling `__wt_page_del_active()` will allow us to proceed and evict the page?

Reply:
You had to lock the `WT_REF` to free the `page_las` structure which makes sense to me, but that means I don't understand the comment in the `WT_REF_DELETED` case:
```
 * We don't have to lock the page, as no thread of
 * control can be running below our locked internal
 * page.
```
Why doesn't the `WT_REF_DELETED` case have to lock before calling `__wt_page_del_active()`?

Reply:
I thought the only path to a page having both lookaside information and being deleted was if it was first written and switched into a lookaside state, and then "fast" deleted.

If we're reverting a lookaside page into the deleted state, I think my understanding is wrong -- can you please clarify that for me?

Reply:
@keithbostic, the difference between the eviction check of a `WT_REF_DELETED` ref and a `WT_REF_LOOKASIDE` ref is that the deleted check is read-only.  Other readers encountering the same deleted ref will make the same decision about whether it is active but we don't actively free the `ref->page_del` structure.

My first attempt tried to free obsolete lookaside history without flipping to locked and encountered errors.  However, that was because of the case where a page has lookaside history but no page image (discussed below).  I'll try backing out the locking (matching the existing comment), to see whether it is really necessary.

The reason a page can go from `WT_REF_LOOKASIDE` to `WT_REF_DELETED` is if all of the entries on the page were deleted, and reconciliation saw the page as empty, but there was history that could not be discarded.  Once that history becomes obsolete, the page is marked deleted directly.  By definition all of the history is obsolete at this point, so there should be no `page_del` structure and hence no need to call `__wt_page_del_active`.  I'll add a check for that.

Reply:
FTR, I think the existing comment is correct and locking is not required for `WT_REF_DELETED` or `WT_REF_LOOKASIDE`.

At some point I was trying to make this work for `WT_REF_LIMBO`, but that is much closer to `WT_REF_MEM` in that a reader can be inside a child in that state, so it isn't safe to evict the parent (we'd have to lock *and* do a hazard check to make that safe and that doesn't seem worth it as we can just evict the limbo page separately, then try the parent again).